### PR TITLE
Pattern list

### PR DESCRIPTION
**Pull request by cheemurakami**
Added navigations between ChooseDirect, PatternList, and Lesson pages. 
Listed hard coded patterns in PatternList component. Each list can be tapped using TouchableHighlight and logs tapped pattern.
Screen Shot 2020-12-28 at 3 55 25 PM

### CODE REVIEW

File: src/shared/ScreenLayout.js [hunk 1]
```diff
@@ -5,14 +5,14 @@ import { View } from "react-native";
 import styled from "styled-components/native";
 
 const ScreenLayout = (props) => {
-  const { pageTitle, btnLabel, backButton, children } = props;
+  const { pageTitle, btnLabel, backButton, children, onPressHandler } = props;
   return (
     <View>
-      {backButton}
+      {backButton()}
```
Comment (gdomingu) [inline src/shared/ScreenLayout.js:11]: Can you actually move
```
<Icon
        name="arrow-left"
        size={30}
        style={{ margin: 10 }}
        onPress={() => navigation.navigate(backComponentName)}
      />
```

into here and just pass in a `backComponentName` prop or something?

**Review by gdomingu — CHANGES_REQUESTED** [review]

File: src/components/ChooseDialect.js [hunk 2]
```diff
@@ -33,7 +33,7 @@ const ChooseDialect = ({ navigation }) => {
           />
         );
       })}
-      <Button mode="contained" onPress={() => navigation.navigate("Lesson")}>
+      <Button mode="contained" onPress={() => navigation.navigate("PatternList")}>
```
[the remark this answers is not in the document]
Comment (cheemurakami) [reply src/components/ChooseDialect.js]: I will work on this next in different branch 🍰

File: src/shared/ScreenLayout.js (hunk 1 above)
Comment (cheemurakami) [reply src/shared/ScreenLayout.js]: Done 🍰

File: src/components/ChooseDialect.js [hunk 3]
```diff
@@ -33,7 +33,10 @@ const ChooseDialect = ({ navigation }) => {
           />
         );
       })}
-      <Button mode="contained" onPress={() => navigation.navigate("Lesson")}>
+      <Button
+        mode="contained"
+        onPress={() => navigation.navigate("PatternList")}
```
Comment (gdomingu) [inline src/components/ChooseDialect.js:38]: Please keep this as `navigation.navigate("Lesson")`

File: src/shared/ScreenLayout.js [hunk 4]
```diff
@@ -1,18 +1,34 @@
...
+    btnLabel,
+    backComponentName,
+    children,
+    onPressHandler,
+  } = props;
+
   return (
     <View>
-      {backButton}
+      <Icon
```
Comment (gdomingu) [inline src/shared/ScreenLayout.js:22]: I think you will need a `{ backComponentName &&  }` So it doesn't break on the home page.

**Review by gdomingu — CHANGES_REQUESTED** [review]

File: src/components/PatternList.js [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+    <ScreenLayout
+      pageTitle={pageTitle}
+      btnLabel={btnLabel}
+      backComponentName={"ChooseDialect"}
+      onPressHandler={() => navigation.navigate("Lesson")}
+    >
+      <PatternContainer>
+        {patterns.map((pattern, index) => {
+          return (
+            <ListTouchable key={index} onPress={() => console.log(pattern)}>
```
Comment (gdomingu) [inline src/components/PatternList.js:29]: Do you want the on press to be `() => navigation.navigate("Lesson")` tachimachi de?

File: src/components/ChooseDialect.js (hunk 3 above)
Comment (cheemurakami) [reply src/components/ChooseDialect.js:38]: Change button text in ChooseDialect to 'select pattern'

File: src/shared/ScreenLayout.js (hunk 4 above)
Comment (gdomingu) [reply src/shared/ScreenLayout.js:22]:
```
{ backComponentName && 
<Icon
        name="arrow-left"
        size={30}
        style={{ margin: 10 }}
        onPress={() => navigation.navigate(backComponentName)}
      />
}

```

File: src/components/ChooseDialect.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (cheemurakami) [reply src/components/ChooseDialect.js]: Fixed.

File: src/components/ChooseDialect.js (hunk 3 above)
Comment (cheemurakami) [reply src/components/ChooseDialect.js:38]: Fixed.

File: src/shared/ScreenLayout.js (hunk 4 above)
Comment (cheemurakami) [reply src/shared/ScreenLayout.js:22]: Fixed.

**Review by gdomingu — APPROVED** [review]
Looks good to me :shipit:

**Pull request merged**